From b2f7a0de9db5aab72c6064a2175309986d70cd07 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Thu, 28 Feb 2013 17:21:50 -0500
Subject: [PATCH 1/2] bbunzip: ignore the -q flag with the decompressors

The -q flag is used in shell scripts for suppressing output.
Have our applets swallow the flag for compatibility.

Reported-by: Mandeep Singh Baines <msb@chromium.org>
Signed-off-by: Mike Frysinger <vapier@gentoo.org>
---
 archival/bbunzip.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/archival/bbunzip.c b/archival/bbunzip.c
index 46f99cf..c444a2c 100644
--- a/archival/bbunzip.c
+++ b/archival/bbunzip.c
@@ -12,8 +12,9 @@ enum {
 	OPT_FORCE      = 1 << 1,
 	/* only some decompressors: */
 	OPT_VERBOSE    = 1 << 2,
-	OPT_DECOMPRESS = 1 << 3,
-	OPT_TEST       = 1 << 4,
+	OPT_QUIET      = 1 << 3,
+	OPT_DECOMPRESS = 1 << 4,
+	OPT_TEST       = 1 << 5,
 };
 
 static
@@ -292,7 +293,7 @@ IF_DESKTOP(long long) int FAST_FUNC unpack_gunzip(transformer_aux_data_t *aux)
 int gunzip_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int gunzip_main(int argc UNUSED_PARAM, char **argv)
 {
-	getopt32(argv, "cfvdtn");
+	getopt32(argv, "cfvdtqn");
 	argv += optind;
 	/* if called as zcat */
 	if (applet_name[1] == 'c')
@@ -330,7 +331,7 @@ IF_DESKTOP(long long) int FAST_FUNC unpack_bunzip2(transformer_aux_data_t *aux)
 int bunzip2_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int bunzip2_main(int argc UNUSED_PARAM, char **argv)
 {
-	getopt32(argv, "cfvdt");
+	getopt32(argv, "cfvdtq");
 	argv += optind;
 	if (applet_name[2] == 'c') /* bzcat */
 		option_mask32 |= OPT_STDOUT;
@@ -398,7 +399,7 @@ IF_DESKTOP(long long) int FAST_FUNC unpack_unlzma(transformer_aux_data_t *aux)
 int unlzma_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int unlzma_main(int argc UNUSED_PARAM, char **argv)
 {
-	IF_LZMA(int opts =) getopt32(argv, "cfvdt");
+	IF_LZMA(int opts =) getopt32(argv, "cfvdtq");
 # if ENABLE_LZMA
 	/* lzma without -d or -t? */
 	if (applet_name[2] == 'm' && !(opts & (OPT_DECOMPRESS|OPT_TEST)))
@@ -423,7 +424,7 @@ IF_DESKTOP(long long) int FAST_FUNC unpack_unxz(transformer_aux_data_t *aux)
 int unxz_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int unxz_main(int argc UNUSED_PARAM, char **argv)
 {
-	IF_XZ(int opts =) getopt32(argv, "cfvdt");
+	IF_XZ(int opts =) getopt32(argv, "cfvdtq");
 # if ENABLE_XZ
 	/* xz without -d or -t? */
 	if (applet_name[2] == '\0' && !(opts & (OPT_DECOMPRESS|OPT_TEST)))
-- 
1.7.12.4

